components/range_cache_memory_engine/src/engine.rs

```diff
@@ -403,6 +407,14 @@ impl RangeCacheMemoryEngine {
             range_manager.record_in_ranges_being_written(write_batch_id, range);
             return RangeCacheStatus::Cached;
         }
+        if range_manager.overlap_with_preferred_range(range) {
```

Comment:
I prefer to handle this in ApplyObserver, but since other similar logic also handled here, I'm ok with this currently.